tests/test_source.py

```diff
@@ -110,3 +110,37 @@ def test_source_multiple_photons_per_mode():
     s = Source(emission_probability=ep)
     svd = s.probability_distribution(2)
     assert_svd_close(svd, dict2svd({"|0>": (1-ep)**2, "|1>": ep*(1-ep)*2, "|2>": ep**2}))
+
+
+def test_source_sample():
+    from perceval.utils import BasicState
+
+    nb_samples = 200
+    source = Source(emission_probability=0.9, multiphoton_component=0.1, losses=0.1, indistinguishability=0.9)
+
+    for bs in [BasicState("|1,1>"), BasicState("|{_:0},{_:1}>")]: # with or without annotations
+        # generate samples directly from the source
+        samples_from_source = source.generate_samples(nb_samples, bs)
+        assert len(samples_from_source) == nb_samples
+        samples_count = {sample: samples_from_source.count(sample)/nb_samples for sample in samples_from_source}
```

Comment:
You can use a conversion method from `utils.conversion` once `samples_from_source` is a `BSSamples`